apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/tool/ServletApiTool.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017, OpenSkywalking Organization All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Project repository: https://github.com/OpenSkywalking/skywalking
+ */
+
+package org.skywalking.apm.agent.core.tool;
+
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * 
+ * @author wendal
+ *
+ */
+public class ServletApiTool {
+
+    // HttpServletResponse.getStatus() is available since Servlet API 3.0
+    protected static boolean RGSA;
+    protected static boolean INITED;
```

Comment:
* Use `volatile` keyword for these two fields. Because you can't be sure the thread module, this field value could be cached by thread, and every thread must cost CPU to initialize the value.